replica/table.cc

```diff
@@ -860,7 +864,13 @@ future<> storage_group::split(sstables::compaction_type_options::split opt) {
         co_return;
     }
 
+    if (_main_cg->empty()) {
+        co_return;
+    }
+    auto holder = _main_cg->async_gate().hold();
     co_await _main_cg->flush();
+    // Waits on sstables produced by repair to be integrated into main set; off-strategy is usually a no-op with tablets.
+    co_await _main_cg->get_compaction_manager().perform_offstrategy(_main_cg->as_table_state(), tasks::task_info{});
```

Comment:
Why is it necessary?

If it is, then it seems we should do it even if _main_cg->empty().

Reply:
let's say repair produced sstable with twcs, they're placed into maintenance set, and off-strategy which integrates into main one, hasn't started yet. This call guarantee that off-strategy start before split, since split works only on main set. Any new sstable produced thereafter is guaranteed to be split either it will be added to main or maintenance set.

_main_cg->empty() means maintenance and main are empty, so there's nothing to be split.

Reply:
> _main_cg->empty() means maintenance and main are empty, so there's nothing to be split.

So _main_cg includes maintenance, we're fine then.